src/Kiota.Builder/Writers/Php/PhpConventionService.cs

```diff
@@ -0,0 +1,236 @@
+                CodeParameterKind.Serializer => $"SerializationWriter {GetParameterName(parameter)}",
+                CodeParameterKind.BackingStore => $"BackingStore {GetParameterName(parameter)}",
+                CodeParameterKind.PathParameters => $"array {GetParameterName(parameter)}",
+                _ => $"{typeString} {GetParameterName(parameter)}"
+
+            };
+            return $"{(!parameter.Optional ? String.Empty : "?")}{parameterSuffix}";
+        }
+        public string GetParameterSignature(CodeParameter parameter, CodeMethod codeMethod)
+        {
+            if (codeMethod?.AccessedProperty != null && codeMethod.AccessedProperty.IsOfKind(CodePropertyKind.AdditionalData))
+            {
+                return "array $value";
+            }
+            
+            return GetParameterSignature(parameter, codeMethod as CodeElement);
+        }
+
+        public string GetParameterDocNullable(CodeParameter parameter, CodeElement codeElement)
+        {
+            var parameterSignature = GetParameterSignature(parameter, codeElement).Trim().Split(' ');
+            return parameter.Optional switch
+            {
+                true => $"{parameterSignature[0].Trim('?')}|null {parameterSignature[1]}",
+                _ => string.Join(' ', parameterSignature)
+            };
+        }
+
+        private static string RemoveInvalidDescriptionCharacters(string originalDescription) => originalDescription?.Replace("\\", "/");
+        public override void WriteShortDescription(string description, LanguageWriter writer)
+        {
+            
+            if (!String.IsNullOrEmpty(description))
+            {
+                writer.WriteLine(DocCommentStart);
+                writer.WriteLine(
+                    $"{DocCommentPrefix}{RemoveInvalidDescriptionCharacters(description)}");
+                writer.WriteLine(DocCommentEnd);
+            }
+        }
+
+        public void AddRequestBuilderBody(bool addCurrentPathProperty, string returnType, LanguageWriter writer, string suffix = default, string additionalPathParameters = default)
+        {
+            writer.WriteLines($"return new {returnType}($this->{RemoveDollarSignFromPropertyName(PathParametersPropertyName)}{suffix}, $this->{RemoveDollarSignFromPropertyName(RequestAdapterPropertyName)});");
+        }
+
+        private static string RemoveDollarSignFromPropertyName(string propertyName)
+        {
+            if (string.IsNullOrEmpty(propertyName) || propertyName.Length < 2)
+            {
+                throw new ArgumentException(nameof(propertyName) + " must not be null and have at least 2 characters.");
+            }
+            
+            return propertyName[1..];
+        }
+
+        public void WritePhpDocumentStart(LanguageWriter writer)
+        {
+            writer.WriteLines("<?php", string.Empty);
+        }
+
+        public void WriteCodeBlockEnd(LanguageWriter writer)
```

Comment:
we now have a CloseBlock method on the langauge writer that does that for you, remove this one, update references